source/mir/random/engine/test.d

```diff
@@ -0,0 +1,236 @@
+module mir.random.engine.test;
+
+import std.typecons : Flag, No;
+
+version(linux) version(X86_64)
+{
+    shared static this()
+    {
+        hasGetRandom = initHasGetRandom;
+    }
+
+    // getrandom was introduced in 3.17
+    private static bool hasGetRandom;
+
+    private auto initHasGetRandom()
+    {
+        import core.sys.posix.sys.utsname;
+        import core.stdc.string : strtok;
+        import core.stdc.stdlib : atoi;
+
+        utsname uts;
+        uname(&uts);
+        char* p = uts.release.ptr;
+
+        auto token = strtok(p, ".");
+        int major = atoi(token);
+        if (major  > 3)
+            return true;
+
+        if (major == 3)
+        {
+            token = strtok(p, ".");
+            if (atoi(token) >= 17)
+                return true;
+        }
+
+        return false;
+    }
+
+    /*
+     * Flags for getrandom(2)
+     *
+     * GRND_NONBLOCK	Don't block and return EAGAIN instead
+     * GRND_RANDOM		Use the /dev/random pool instead of /dev/urandom
+     */
+    private enum GRND_NONBLOCK = 0x0001;
+    private enum GRND_RANDOM = 0x0002;
+    private enum GETRANDOM = 318;
+
+    private size_t syscall(size_t ident, size_t n, size_t arg1, size_t arg2)
+    {
+        size_t ret;
+
+        asm
+        {
+            mov RAX, ident;
+            mov RDI, n[RBP];
+            mov RSI, arg1[RBP];
+            mov RDX, arg2[RBP];
+            syscall;
+            mov ret, RAX;
+        }
```

Comment:
I got the inspiration to use this `syscall` style from here: https://github.com/kubo39/syscall-d/blob/master/source/syscalld/arch/syscall_x86_64.d
Hence also the `synchronized` statements, but afaict we don't need it - e.g.:
https://wiki.dlang.org/LDC_inline_assembly_expressions#X86-64